olive/engine/engine.py

```diff
@@ -87,6 +90,38 @@ def __init__(
         else:
             self.target = LocalSystem()
 
+        if execution_providers is None:
+            execution_providers = self._config.execution_providers
+
+        # verfiy the AzureML system have specified the execution providers
+        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
+        if self.target.system_type == SystemType.AzureML and execution_providers is None:
+            raise ValueError("AzureMLSystem requires execution providers to be specified.")
+        elif execution_providers is None:
+            if self.target.system_type == SystemType.Local:
+                execution_providers = self.target.get_supported_execution_providers()
+            else:
+                # for docker system and python system, we default use CPUExecutionProvider
```

Comment:
python system also has a similar method to `get_supported_execution_providers` called `get_available_eps`. We can rename `get_available_eps` and use it instead of default to cpu ep.